### PR TITLE
Add node 6

### PR DESCRIPTION
**Pull request by thailekha**
### Node 6 support:
- Update engines to node 6.9
- Change `^` to `~`, and `*` to specific version
- update grunt-fh-build
- Replace `geoip` with `geoip-lite` due to `node-gyp` error during `npm install`
- Add `countries-list` to fill in missing data from geoip-lite's output
- Update Jenkinfiles for Wendy build

JIRA: https://issues.jboss.org/browse/RHMAP-16212
https://issues.jboss.org/browse/RHMAP-16524

### CODE REVIEW

**Comment by thailekha** [comment]
retest this please

File: fh-messaging/lib/fh_filter_man.js [hunk 1]
```diff
@@ -176,7 +176,6 @@ var filterOneAsync = function(message, cb) {
 function FilterManager(config, logger) {
```
Comment (fh-qube) [inline fh-messaging/lib/fh_filter_man.js]: INFO Rename this 'FilterManager' function to match the regular expression ^[a-z][a-zA-Z0-9]*$ [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS100)

**Comment by fh-qube** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 1 major
* INFO 1 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Refactor this code to not nest more than 3 if/for/while/switch/try statements.](https://github.com/feedhenry/fh-messaging/blob/8dd321ce65f5882002065c1d6506b751d7aaa1fb/fh-messaging/lib/fh_filter_man.js#L79) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ANestedIfDepth)

**Comment by fh-qube** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 1 major
* INFO 1 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Refactor this code to not nest more than 3 if/for/while/switch/try statements.](https://github.com/feedhenry/fh-messaging/blob/67a0adf1a100ac00a517285dccdbf40410b55b79/fh-messaging/lib/fh_filter_man.js#L79) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ANestedIfDepth)

**Comment by fh-qube** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 1 major
* INFO 1 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Refactor this code to not nest more than 3 if/for/while/switch/try statements.](https://github.com/feedhenry/fh-messaging/blob/3649f5457421ee0be92afb74b44843453a52cdb1/fh-messaging/lib/fh_filter_man.js#L79) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ANestedIfDepth)

**Comment by fh-qube** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 1 major
* INFO 1 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Refactor this code to not nest more than 3 if/for/while/switch/try statements.](https://github.com/feedhenry/fh-messaging/blob/760c928a5538ef8806d09b62ea0fce2d069d711d/fh-messaging/lib/fh_filter_man.js#L79) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ANestedIfDepth)

**Comment by thailekha** [comment]
@rachael-oregan yes since the dependencies are updated

File: fh-messaging/lib/fh_filter_man.js [hunk 2]
```diff
@@ -55,18 +51,22 @@ var filterFuncs = {
...
-      });
-    } else {
-      return callback(null, filtered);
+      var geo = geoip.lookup(ip);
+      mLogger.debug("Result from geoip: " + JSON.stringify(geo));
+      if (!geo || !geo.country) {
+        mLogger.warn("Error from geoip lookup, continuing anyway with no country for ip: " + ip);
+      } else {
+        filtered.country = {
+          continent_code: countries[geo.country].continent,
```
Comment (pb82) [inline fh-messaging/lib/fh_filter_man.js]: What if the country is not in the list / has a different code?

**Comment by fh-qube** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 1 major
* INFO 1 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Refactor this code to not nest more than 3 if/for/while/switch/try statements.](https://github.com/feedhenry/fh-messaging/blob/e756d65898e8db657dd291ed22d08e8d7ebd89e6/fh-messaging/lib/fh_filter_man.js#L79) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ANestedIfDepth)

File: fh-messaging/lib/fh_filter_man.js (hunk 2 above)
Comment (thailekha) [inline fh-messaging/lib/fh_filter_man.js]: Updated to check if the country code is in the list, thanks! Is it ok now?

File: fh-messaging/migrate/post_0.7.0.js [hunk 3]
```diff
@@ -26,7 +25,14 @@ db.on('tearUp', function (err) {
                 var ii, il;
                 console.log('entries: ' + items.length);
                 for (ii = 0, il = items.length; ii < il; ii += 1) {
-                  items[ii].country = country.lookupSync(items[ii].ipAddress);
+                  var geo = geoip.lookup(items[ii].ipAddress).country;
```
Comment (thailekha) [inline fh-messaging/migrate/post_0.7.0.js]: @pb82 should I add an if-statement here?

File: fh-metrics/package.json [hunk 4]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "fh-metrics",
   "description": "FeedHenry Metrics Server",
-  "version": "3.0.6-BUILD-NUMBER",
+  "version": "3.0.7-BUILD-NUMBER",
```
Comment (laurafitzgerald) [inline fh-metrics/package.json]: I notice you bumped to 3.1.0 on fh-messaging. Any reason for the difference?

Comment (thailekha) [inline fh-metrics/package.json]: On fh-messaging, I changed package.json and the code, so I bumped the minor. On fh-metrics, I just changed package.json, so I bumped the patch only.

File: npm-shrinkwrap.json [hunk 5]
```diff
@@ -1,11 +1,11 @@
 {
   "name": "fh-messaging-acceptance-tests",
-  "version": "0.2.2",
+  "version": "0.2.3",
   "dependencies": {
     "moment": {
-      "version": "2.15.2",
-      "from": "moment@>=2.10.6 <3.0.0",
-      "resolved": "https://registry.npmjs.org/moment/-/moment-2.15.2.tgz"
+      "version": "2.10.6",
```
Comment (jcesarmobile) [inline npm-shrinkwrap.json]: as it was "moment": "^2.10.6" it got "2.15.2" last time this was generated.
Now it's "~2.10.6", so we have downgraded the version.

We should keep the same versions we were using, and use that version with ~, not just replace `^` with `~`

**Comment by fh-qube** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 1 major
* INFO 1 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Refactor this code to not nest more than 3 if/for/while/switch/try statements.](https://github.com/feedhenry/fh-messaging/blob/21137e8adf5a9a55ac67142d513b5fc62749a19d/fh-messaging/lib/fh_filter_man.js#L79) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ANestedIfDepth)

**Comment by fh-qube** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 1 major
* INFO 1 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Refactor this code to not nest more than 3 if/for/while/switch/try statements.](https://github.com/feedhenry/fh-messaging/blob/5b6f976facd1021e01405224be331a799c7a614a/fh-messaging/lib/fh_filter_man.js#L79) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ANestedIfDepth)

**Comment by fh-qube** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 1 major
* INFO 1 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Refactor this code to not nest more than 3 if/for/while/switch/try statements.](https://github.com/feedhenry/fh-messaging/blob/d65f27a1673b0250e899434454ac655a7fc328dc/fh-messaging/lib/fh_filter_man.js#L79) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ANestedIfDepth)

**Comment by fh-qube** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 1 major
* INFO 1 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Refactor this code to not nest more than 3 if/for/while/switch/try statements.](https://github.com/feedhenry/fh-messaging/blob/407327661af22177dcc8efee6beb4025e850ec06/fh-messaging/lib/fh_filter_man.js#L79) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ANestedIfDepth)

**Comment by fh-qube** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 1 major
* INFO 1 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Refactor this code to not nest more than 3 if/for/while/switch/try statements.](https://github.com/feedhenry/fh-messaging/blob/b4a4b43be7f098641d69929b641758323f0f1ad9/fh-messaging/lib/fh_filter_man.js#L79) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ANestedIfDepth)

**Comment by thailekha** [comment]
retest this please

**Comment by fh-qube** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 1 major
* INFO 1 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Refactor this code to not nest more than 3 if/for/while/switch/try statements.](https://github.com/feedhenry/fh-messaging/blob/b4a4b43be7f098641d69929b641758323f0f1ad9/fh-messaging/lib/fh_filter_man.js#L79) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ANestedIfDepth)

**Comment by thailekha** [comment]
retest this please

**Comment by fh-qube** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 1 major
* INFO 1 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Refactor this code to not nest more than 3 if/for/while/switch/try statements.](https://github.com/feedhenry/fh-messaging/blob/b4a4b43be7f098641d69929b641758323f0f1ad9/fh-messaging/lib/fh_filter_man.js#L79) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ANestedIfDepth)

**Comment by fh-qube** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 1 major
* INFO 1 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Refactor this code to not nest more than 3 if/for/while/switch/try statements.](https://github.com/feedhenry/fh-messaging/blob/bec1ed965b378da3dd041a16a5c4603d84df52c3/fh-messaging/lib/fh_filter_man.js#L79) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ANestedIfDepth)

**Comment by fh-qube** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 1 major
* INFO 1 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Refactor this code to not nest more than 3 if/for/while/switch/try statements.](https://github.com/feedhenry/fh-messaging/blob/3a2d952139c6564ddddbb3b18144f0d330c7ecf2/fh-messaging/lib/fh_filter_man.js#L79) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ANestedIfDepth)

**Comment by fh-qube** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 1 major
* INFO 1 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Refactor this code to not nest more than 3 if/for/while/switch/try statements.](https://github.com/feedhenry/fh-messaging/blob/0b2f60600254d60fa859ea0a235caeec35e981bc/fh-messaging/lib/fh_filter_man.js#L79) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ANestedIfDepth)

**Comment by grdryn** [comment]
Hi! Closing this PR as it seems stale. If it's still relevant, please rebase and feel free to re-open. Thanks!

**Pull request closed without merging**